src/Command/RemoveOldImagesCommand.php

```diff
@@ -0,0 +1,319 @@
+     * Limit by batch size.
+     *
+     * @return number Number of removed records from database
+     */
+    private function deletePostCommentsImages(OutputInterface $output): int
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $timeAgo = new \DateTime("-{$this->monthsAgo} months");
+
+        $query = $queryBuilder
+            ->select('c')
+            ->from(PostComment::class, 'c')
+            ->where(
+                $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->lt('c.createdAt', ':timeAgo'),
+                    $queryBuilder->expr()->neq('i.id', 1),
+                    $queryBuilder->expr()->isNotNull('c.apId'),
+                    $this->noActivity ? $queryBuilder->expr()->eq('c.upVotes', 0) : null,
+                    $this->noActivity ? $queryBuilder->expr()->eq('c.favouriteCount', 0) : null
+                )
+            )
+            ->innerJoin('c.image', 'i')
+            ->orderBy('c.id', 'ASC')
+            ->setParameter('timeAgo', $timeAgo)
+            ->setMaxResults($this->batchSize)
+            ->getQuery();
+
+        $comments = $query->getResult();
+
+        foreach ($comments as $comment) {
+            $output->writeln(\sprintf('Deleting image from post comment ID: %d, with ApId: %s', $comment->getId(), $comment->getApId()));
+            $this->postCommentManager->detachImage($comment);
+        }
+
+        // Return total number of elements deleted
+        return \count($comments);
+    }
+
+    /**
+     * Delete user avatar and user cover images. Check ap_fetched_at column for the age.
+     * Limit by batch size.
+     *
+     * @return number Number of removed records from database
+     */
+    private function deleteUsersImages(OutputInterface $output): int
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $timeAgo = new \DateTime("-{$this->monthsAgo} months");
+
+        $query = $queryBuilder
+            ->select('u')
+            ->from(User::class, 'u')
+            ->where(
+                $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->orX(
+                        $queryBuilder->expr()->isNotNull('u.avatar'),
+                        $queryBuilder->expr()->isNotNull('u.cover')
+                    ),
+                    $queryBuilder->expr()->lt('u.apFetchedAt', ':timeAgo'),
+                    $queryBuilder->expr()->isNotNull('u.apId')
+                )
+            )
+            ->orderBy('u.apFetchedAt', 'ASC')
+            ->setParameter('timeAgo', $timeAgo)
+            ->setMaxResults($this->batchSize)
+            ->getQuery();
+
+        $users = $query->getResult();
+
+        foreach ($users as $user) {
+            $output->writeln(\sprintf('Deleting image from username: %s', $user->getUsername()));
+            $this->userManager->detachCover($user);
+            $this->userManager->detachAvatar($user);
```

Comment:
I think it would be a good practice to check for the the variables, even if it is already checked down the line
```suggestion
            if (null !== $user->cover) {
                $this->userManager->detachCover($user);
            }
            if (null !== $user->avatar) {
                $this->userManager->detachAvatar($user);
            }
```